pkg/roll/run_update.go

```diff
@@ -152,17 +152,7 @@ ROLL_LOOP:
 				break
 			}
 
-			if newNodes.Desired > newNodes.Healthy {
-				// we assume replication controllers do in fact "work", ie healthy
-				// and current converge towards desired
-				// in this case, they have not yet met, so we block
-				u.logger.WithFields(logrus.Fields{
-					"old": oldNodes,
-					"new": newNodes,
-				}).Debugln("Blocking for more healthy new nodes")
-				break
-			}
-			if newNodes.Desired >= u.DesiredReplicas {
+			if newNodes.Desired >= u.DesiredReplicas && newNodes.Healthy >= u.MinimumReplicas {
 				// note that we only exit the loop AFTER the desired nodes
```

Comment:
there's a second sentence beginning with "normal execution of this rollAlgorithm..." which is just explaining why the condition is `newNodes.Desired >= u.DesiredReplicas` rather than `>`, that part I understand.